src/ec/curve25519/ops.rs

```diff
@@ -84,16 +81,32 @@ impl Point {
     }
 
     pub fn to_bytes(&self) -> [u8; ELEM_LEN] {
-        let mut bytes = [0u8; ELEM_LEN];
-        unsafe { GFp_x25519_ge_tobytes(&mut bytes, self); }
+        point_to_bytes(&self.x, &self.y, &self.z)
+    }
+}
 
-        bytes
+fn point_to_bytes(x: &Elem, y: &Elem, z: &Elem) -> [u8; ELEM_LEN] {
+    let mut recip = [0; ELEM_LIMBS];
+    let mut x_over_z = [0; ELEM_LIMBS];
+    let mut y_over_z = [0; ELEM_LIMBS];
+    let mut bytes = [0; ELEM_LEN];
+
+    unsafe {
+        GFp_fe_invert(&mut recip, z);
+        GFp_fe_mul(&mut x_over_z, x, &recip);
+        GFp_fe_mul(&mut y_over_z, y, &recip);
+        GFp_fe_tobytes(&mut bytes, &y_over_z);
     }
+    bytes[31] ^= unsafe { GFp_fe_isnegative(&x_over_z) << 7 };
```

Comment:
Determining whether x_over_z is negative needs to be done in constant time, but merging its result into `bytes[31]` doesn't need to be constant time, right?

Maybe it would be better to do the constant-time operation in a separate statement from the non-constant-time operation? Then every statement would be constant-time or not? Then we could also document that it isn't necessary for the `bytes ^= ...` to be constant time even though it is necessary to calculate both inputs of the operation in constant time.